src/main/webapp/app/view/project/admin/locations/project-location-update.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpResponse } from '@angular/common/http';
+import { FormBuilder, Validators } from '@angular/forms';
+import { Observable } from 'rxjs';
+
+import { Project } from 'app/shared/model/project.model';
+import { Location } from 'app/shared/model/location.model';
+
+import { LocationService } from 'app/entities/location/location.service';
+import { Responsibility } from 'app/shared/model/responsibility.model';
+import { ResponsibilityService } from 'app/entities/responsibility/responsibility.service';
+
+@Component({
+  selector: 'app-location-update',
+  templateUrl: './project-location-update.component.html',
+  styleUrls: ['./project-location-update.component.scss'],
+})
+export class ProjectLocationUpdateComponent implements OnInit {
+  isSaving = false;
+  isNew = false;
+  project?: Project;
+  responsibilities: Responsibility[] = [];
+  //filteredResponsibilities?: Observable<Responsibility[]>;
+  editForm = this.fb.group({
+    id: [],
+    name: [null, [Validators.required, Validators.maxLength(50)]],
+    responsibility: [null, [Validators.required]],
+    project: [],
+  });
+
+  constructor(protected locationService: LocationService, private fb: FormBuilder, private responsibilityService: ResponsibilityService) {}
+
+  ngOnInit(): void {
+    /* this.filteredResponsibilities = this.editForm.get('responsibility')!.valueChanges.pipe(
+       startWith(''),
+       map(value => (typeof value === 'string' ? value : value.name)),
+       map(name => this.filter(name))
+     );*/
+  }
+
+  filter(name: string): Responsibility[] {
+    const filterValue = name.toLowerCase();
+
+    if (!this.responsibilities) {
+      return [] as Responsibility[];
+    }
+
+    return this.responsibilities.filter(responsibility => responsibility.name!.toLowerCase().includes(filterValue));
+  }
+
+  public updateForm(project: Project, location: Location): void {
+    this.isNew = !location.id;
+    this.responsibilityService.findAllByProject(project).subscribe(responsibilities => {
+      this.responsibilities = responsibilities.body || [];
+    });
+
+    this.editForm.patchValue({
```

Comment:
fyi you could also dereference the `location` object like this `const {id, name, responsibility} = location;`. that allows you to patch the value as following `this.editForm.patchValue({id, name, responsibility, project});`.
(I am not sure if your linting rules allow it, but you could try if you want to ;) )